lib/filters/index.js

```diff
@@ -75,6 +73,29 @@ module.exports = ruleSource => {
         }
       }
 
+      //hack to support nodejs gitlab client
+      const GITLAB_DUMMY_QUERY =
+        'private_token=6E6ACA18-70B8-4FE3-A602-611209920236';
+      if (validParams && querystring) {
+        const paramsOptions = querystring.split('&');
+
+        // validate against the query params
+        const isValid = paramsOptions.every((queryParam) => {
+          if (queryParam.includes('private_token')) {
+            return true;
+          }
+
+          return validParams.some(({ param, value }) => {
+            return (queryParam == `${param}=${value}`);
+          });
+        });
+
+        if (!isValid) {
+          return false;
+        }
+
+      }
```

Comment:
Instead of this validation (which btw has a problem because if `ref=master` appears, than bad paths are also allowed?), new `"validParams"` object allows this:
```
if (validParams && querystring) {
  return querystring.split('&').all(pair => {
    const [key, val] = pair.split('=');
    if (!validParams[key]) {
      return true;
    }
    return validParams[key].includes(val);
  });
}
```

Not sure if the code above works :)

Plus, it surely doesn't work for nested files, that would be important to fix.